provenance/core.py

```diff
@@ -84,6 +90,19 @@ def fn_info(f):
             valid_serializer = valid_serializer and serializer in s.serializers
             if not valid_serializer:
                 break
+    elif info['archive_file']:
+        serializer = metadata.get('serializer', 'file')
+        if serializer != 'file':
+            raise ValueError("With 'archive_file' set True the only valid 'serializer' is 'file'")
+        if metadata.get('dump_kwargs') is not None:
+            raise ValueError("With 'archive_file' set True you may not specify any dump_kwargs.")
+        if metadata.get('load_kwargs') is not None:
+            raise ValueError("With 'archive_file' set True you may not specify any load_kwargs.")
+        info['serializer'] = 'file'
+        info['load_kwargs'] = metadata.get('load_kwargs', {})
+        info['dump_kwargs'] = metadata.get('dump_kwargs',
+                                           {'delete_original': metadata['delete_original_file']})
```

Comment:
It is guaranteed to always exist since it is listed as an actual parameter in the decorator whereas the others came along via the `kwargs`.